### PR TITLE
server: return after respond_to_request() for EROFS

### PR DESCRIPTION
**Pull request by xiexingguo**
Respond_to_request will do cleanup jobs, thus we shall
stall and exit under this case.

Signed-off-by: xie xingguo <xie.xingguo@zte.com.cn>

### CODE REVIEW

File: src/mds/Server.cc [hunk 1]
```diff
@@ -3652,8 +3652,8 @@ void Server::handle_client_file_readlock(MDRequestRef& mdr)
     break;
 
   default:
-    dout(10) << "got unknown lock type " << checking_lock.type
-	     << ", dropping request!" << dendl;
+    dout(10) << "got unknown lock type " << checking_lock.type << dendl;
+    respond_to_request(mdr, -EINVAL);
```
Comment (jcsp) [inline src/mds/Server.cc]: What is the client behaviour in this case?

**Comment by jcsp** [comment]
The actual EROFS part of this is fine, not so sure about the unknown lock type change until we see how a client would actually handle EINVAL

**Comment by xiexingguo** [comment]
@jcsp Thanks for the careful review, john. Seems it passed the test, but let me know if you want me to drop that change.

**Comment by jcsp** [comment]
On reflection returning EINVAL to clients in the unknown lock case is probably okay, since it indicates bad behavour from the client (and I don't mind upsetting the client with a return code it might not handle gracefully in that case).

**Comment by jcsp** [comment]
http://pulpito.ceph.com/jspray-2016-06-14_13:37:22-fs-wip-jcsp-testing-20160614-distro-basic-mira/

**Comment by gregsfortytwo** [comment]
FYI: https://github.com/ceph/ceph/pull/11985 had to revert the Mutation::_dump() patch.

**Comment by xiexingguo** [comment]
> FYI: #11985 had to revert the Mutation::_dump() patch.

Thanks for your reminder, Greg. And sorry for that:-(

**Pull request merged**